lib/KDateRange/ValidationMachine.js

```diff
@@ -0,0 +1,140 @@
+import { createMachine, assign } from 'xstate';
+import { isAfter, startOfDay, isBefore } from 'date-fns';
+import validationConstants, { DATE_FMT } from './validationConstants';
+
+/** Returns if the given prop is equal to the placeholder */
+function isPlaceholder(dateStr) {
+  return dateStr === DATE_FMT;
+}
+
+/** Returns if the given prop matches the constant dateFormat RegExp pattern */
+const isCorrectFormat = dateStr => {
+  return dateFormat.test(dateStr) || dateStr === DATE_FMT;
+};
+
+/** Returns if the end date is after the start date */
+const isEndDateAfterStart = (startDate, endDate) => {
+  const [startDay, startMonth, startYear] = startDate.split('/');
+  const newStartDate = startOfDay(new Date(startYear, startMonth - 1, startDay));
+
+  const [endDay, endMonth, endYear] = endDate.split('/');
+  const newEndDate = startOfDay(new Date(endYear, endMonth - 1, endDay));
+  return isAfter(newStartDate, newEndDate);
+};
+
+/** Returns if the given date string is after the last allowed date */
+const isDateAfterLastAllowed = (dateStr, lastAllowedDate) => {
+  const [day, month, year] = dateStr.split('/');
+  const newDate = startOfDay(new Date(year, month - 1, day));
+  return isAfter(newDate, lastAllowedDate);
+};
+
+/** Returns if the given date string is before the first allowed date */
```

Comment:
For these functions above, any reason not to also document the arguments, like you've done for `validate`?

Reply:
There is no reason not to. I can see how documenting the arguments makes things clearer, so I'll update these functions.